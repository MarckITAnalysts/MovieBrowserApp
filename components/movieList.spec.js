import React from 'react';
import renderer from 'react-test-renderer';
import MovieList from './movieList';


const mockedUsedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => mockedUsedNavigate,
}));

const data = [
  {
    "adult": false,
    "backdrop_path": "/pWsD91G2R1Da3AKM3ymr3UoIfRb.jpg",
    "genre_ids": [
      878,
      28,
      18
    ],
    "id": 933131,
    "original_language": "ko",
    "original_title": "황야",
    "overview": "After a deadly earthquake turns Seoul into a lawless badland, a fearless huntsman springs into action to rescue a teenager abducted by a mad doctor.",
    "popularity": 2140.707,
    "poster_path": "/zVMyvNowgbsBAL6O6esWfRpAcOb.jpg",
    "release_date": "2024-01-26",
    "title": "Badland Hunters",
    "video": false,
    "vote_average": 6.742,
    "vote_count": 433
  },
]

describe('movie list test cases', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<MovieList title="Now Playing" hideSeeAll={false} data={data} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
