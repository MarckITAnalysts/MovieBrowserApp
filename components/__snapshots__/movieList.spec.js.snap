// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movie list test cases renders correctly 1`] = `
<View
  style={
    [
      {
        "marginBottom": 32,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "marginLeft": 16,
          "marginRight": 16,
        },
        {
          "flexDirection": "row",
        },
        {
          "justifyContent": "space-between",
        },
        {
          "alignItems": "center",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#fff",
          },
          {
            "fontSize": 18,
            "lineHeight": 28,
          },
        ]
      }
    >
      Now Playing
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 18,
              "lineHeight": 28,
            },
            {
              "color": "#FFFFFF",
            },
          ]
        }
      >
        See All
      </Text>
    </View>
  </View>
  <RCTScrollView
    contentContainerStyle={
      {
        "paddingHorizontal": 15,
      }
    }
    horizontal={true}
    showsHorizontalScrollIndicator={false}
    style={
      [
        {
          "marginTop": 16,
        },
      ]
    }
  >
    <View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "marginRight": 16,
            },
          ]
        }
      >
        <Image
          source={
            {
              "uri": "https://image.tmdb.org/t/p/w185/zVMyvNowgbsBAL6O6esWfRpAcOb.jpg",
            }
          }
          style={
            [
              {
                "borderBottomLeftRadius": 24,
                "borderBottomRightRadius": 24,
                "borderTopLeftRadius": 24,
                "borderTopRightRadius": 24,
              },
              {
                "height": 293.48,
                "width": 247.5,
              },
            ]
          }
        />
        <Text
          style={
            [
              {
                "color": "#d4d4d4",
              },
              {
                "marginLeft": 4,
              },
              [
                {
                  "marginTop": 4,
                },
              ],
            ]
          }
        >
          Badland Hunter...
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
